backend/inspirehep/assign/views.py

```diff
@@ -49,7 +49,12 @@ def assign_papers(
 ):
     author_bai = get_values_for_schema(to_author_record["ids"], "INSPIRE BAI")[0]
     for record in author_papers:
-        lit_author = get_author_by_recid(record, from_author_recid)
+        try:
+            lit_author = get_author_by_recid(record, from_author_recid)
+        # it might happen that stub author BAI already existed
+        # and we already assigned papers to his profile.
+        except StopIteration:
+            continue
```

Comment:
This is a workaround for #2177. I don't think we should ignore the issue but rather find and fix the root cause. How is this related to this issue?

Reply:
I am not sure if that's a workaround for another issue cause this one is about `get_author_by_recid` and the other one is about BAI. In the case that I try to fix the following problem, due to which my test for commiting was failing:   
we create a new stub author and [here](https://github.com/inspirehep/inspirehep/blob/39bbacf372b720b6cfed7ce577f74abf72256dc5/backend/inspirehep/records/api/authors.py#L91) we assign reference to this new author so if we try to get an old author by recid it's not valid anymore and we get an error.

Reply:
You're right of course about the issue being different, I got confused by the `StopIteration` that made it look superficially similar. I don't understand the failure mode however. Let's discuss it.